c8/function/function_interface.py

```diff
@@ -0,0 +1,324 @@
+        :rtype: dict
+        """
+
+        query_parameters = {
+            "type": worker_type,
+            "name": name,
+            "queryWorkerName": query_worker_name,
+            "environment": environment,
+        }
+        if worker_type != "akamai":
+            query_parameters["type"] = worker_type
+        if environment != "production":
+            query_parameters["environment"] = environment
+
+        request = Request(
+            method="post",
+            endpoint="/function/generate",
+            params=query_parameters,
+            data={},
+        )
+
+        return self.execute(request, FunctionServerError)
+
+    def deploy_stream_publisher_to_edge_worker(
+        self,
+        name,
+        stream_worker_name,
+        stream_name,
+        worker_type="akamai",
+        environment="PRODUCTION",
+    ):
+        """
+        Deploy the stream publisher to the edge worker environment.
+        Note: The edge worker activation is delayed while the environment is set up.
+        Use the get edge function information API to get the activation status of
+        the edge worker.
+
+        :param worker_type: Key of the data
+        :type worker_type: str
+        :param name: Name of the edge worker
+        :type name: str
+        :param stream_worker_name: Name of the Macrometa stream worker
+        :type stream_worker_name: str
+        :param stream_name: Name of the Macrometa stream
+        :type stream_name: str
+        :param environment: The environment in which the edge worker is activated.
+        STAGING | PRODUCTION
+        :type environment: str
+
+        :returns: Returns response in format {"code": xx, "error": xx "result": xx}
+        :rtype: dict
+        """
+        query_parameters = {
+            "type": worker_type,
+            "name": name,
+            "streamWorkerName": stream_worker_name,
+            "streamName": stream_name,
+            "environment": environment,
+        }
+
+        if worker_type != "akamai":
+            query_parameters["type"] = worker_type
+        if environment != "production":
+            query_parameters["environment"] = environment
+
+        request = Request(
+            method="post",
+            endpoint="/function/generate/publisher",
+            params=query_parameters,
+            data={},
+        )
+
+        return self.execute(request, FunctionServerError)
+
+    def deploy_stream_adhoc_query_to_edge_worker(
+        self, name, stream_worker_name, worker_type="akamai", environment="PRODUCTION"
+    ):
+        """
+        Deploy the stream adhoc query to the edge worker environment. Note: The edge worker
+        activation is delayed while the environment is set up. Use the get function
+        worker information API to get the activation status of the edge worker.
+
+        :param worker_type: Key of the data
+        :type worker_type: str
```

Comment:
I very minor thing... I see most of the endpoints have worker_type declared as the third param but in docstring we described it first , maybe we can keep it in same order as declared in function

Reply:
Yea, I was using API reference but I should look at method call 👍